crates/router/src/core/payment_methods/cards.rs

```
@@ -2840,11 +2841,19 @@ pub async fn update_payment_method_connector_mandate_details(
     key_store: &domain::MerchantKeyStore,
     db: &dyn db::StorageInterface,
     pm: domain::PaymentMethod,
-    connector_mandate_details: Option<serde_json::Value>,
+    connector_mandate_details: Option<diesel_models::CommonMandateReference>,
     storage_scheme: MerchantStorageScheme,
 ) -> errors::CustomResult<(), errors::VaultError> {
+    let connector_mandate_details_value = connector_mandate_details
+        .map(|common_mandate| {
+            common_mandate
+                .get_mandate_details_value()
+                .map_err(|_| errors::VaultError::UpdateInPaymentMethodDataTableFailed)
```

Avoid discarding the error, log the error.